pandas/core/dtypes/cast.py

```diff
@@ -604,6 +605,41 @@ def conv(r, dtype):
     return [conv(r, dtype) for r, dtype in zip(result, dtypes)]
 
 
+def maybe_convert_for_categorical(categories, dtype):
+    """Convert ``categories`` depending on ``dtype``.
+
+    Converts to numeric, datetime, or timedelta types, when ``dtype`` is
+    a CategoricalDtype with known, non-object categories.
+
+    Parameters
+    ----------
+    categories : array-like
+    type : CategoricalDtype
+
+    Returns
+    -------
+    new_categories : array or Index
+
+    Examples
+    --------
+    >>> maybe_convert_for_categorical(['1', '2'], CategoricalDtype([1, 2]))
+    array([  1,  2])
+    >>> maybe_convert_for_categorical([1, 'a'], CategoricalDtype([1, 2]))
+    array([  1.,  nan])
+    """
+    if isinstance(dtype, CategoricalDtype) and dtype.categories is not None:
```

Comment:
in reaility this just an Index routine maybe

``Index(categories).coerce_to_dtype(dtype.categories)``

and if the ifisinstance(dtype, ....) logic can be in ``Categorical_from_inferred....``

Reply:
see my comment below, you can simply fold this in to ``Categorical._from_infererd_categories`` / not averse to making the inside of this an Index routine though (as its just coercing on the index type).